src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -30,38 +30,464 @@ using namespace dataRepository;
 MeshLevel::MeshLevel( string const & name,
                       Group * const parent ):
   Group( name, parent ),
-  m_nodeManager( groupStructKeys::nodeManagerString, this ),
-  m_edgeManager( groupStructKeys::edgeManagerString, this ),
-  m_faceManager( groupStructKeys::faceManagerString, this ),
-  m_elementManager( groupStructKeys::elemManagerString, this ),
-  m_embSurfNodeManager( groupStructKeys::embSurfNodeManagerString, this ),
-  m_embSurfEdgeManager( groupStructKeys::embSurfEdgeManagerString, this )
+  m_nodeManager( new NodeManager(groupStructKeys::nodeManagerString, this) ),
+  m_edgeManager( new EdgeManager(groupStructKeys::edgeManagerString, this) ),
+  m_faceManager( new FaceManager(groupStructKeys::faceManagerString, this) ),
+  m_elementManager( new ElementRegionManager(groupStructKeys::elemManagerString(), this) ),
+  m_embSurfNodeManager( new EmbeddedSurfaceNodeManager( groupStructKeys::embSurfNodeManagerString, this) ),
+  m_embSurfEdgeManager( new EdgeManager( groupStructKeys::embSurfEdgeManagerString, this) ),
+  m_isShallowCopy(false)
+{
+
+  registerGroup( groupStructKeys::nodeManagerString, m_nodeManager );
+
+  registerGroup( groupStructKeys::edgeManagerString, m_edgeManager );
+
+
+  registerGroup< FaceManager >( groupStructKeys::faceManagerString, m_faceManager );
+  m_faceManager->nodeList().setRelatedObject( *m_nodeManager );
+
+
+  registerGroup< ElementRegionManager >( groupStructKeys::elemManagerString(), m_elementManager );
+
+  registerGroup< EdgeManager >( groupStructKeys::embSurfEdgeManagerString, m_embSurfEdgeManager );
 
+  registerGroup< EmbeddedSurfaceNodeManager >( groupStructKeys::embSurfNodeManagerString, m_embSurfNodeManager );
+
+  registerWrapper< integer >( viewKeys.meshLevel );
+}
+
+
+MeshLevel::MeshLevel( string const & name,
+                      Group * const parent,
+                      MeshLevel & source ):
```

Comment:
why do you need to make a shallow copy like this? Like, in what situation would you want 2 `MeshLevels` which are identical?

Reply:
This is just a temporary fix. What needs to happen in the end is that we interface with the `MeshLevel` which manages its own relationship with the base discretization/geometry. So right now we don't want to have physics solvers working on the `baseDiscretization`, so we make a shallow copy because we know that this is how it will be in the future.